### PR TITLE
Material Tree JEI page returns

### PR DESCRIPTION
**Pull request by kumquat-ir**
**What:**
TL;DR: #48 but I fixed some git stuff
Adds a new JEI page that shows a flowchart-esque view of common prefixes that a `Material` (with a `DustProperty`) has and some information about it. This reduces the annoyance of trying to find out what forms a material has, how to make them, and what to do with them.

**Additional info:**
There will be a re-layout soon, some stuff is broken by different assumptions in the new material system

Adds new helper class DrawableRegistry, which is a glorified HashMap<String, IDrawable> for reducing boilerplate when registering a large number of IDrawables, such as the arrows on the flowchart

Currently is set to register ingredients as both inputs and outputs, so it shows up on left-click and right-click, for ease of use

Generic tooltips showing the most common ways to make a prefix will be added soon, unless there's some way to easily get the actual recipes (which I heavily doubt)

**Possible compatibility issue:**
None

### CODE REVIEW

**Comment by ALongStringOfNumbers** [comment]
A bit of weird formatting for indentations

**Review by DStrand1 — APPROVED** [review]
